src/main/webapp/js/common/react/components/DataFetcher.js

```diff
@@ -0,0 +1,71 @@
+import checkTesting from '../../../checkTesting.js';
+import {basicErrorHandling} from '../../../fetch_handler.js';
+checkTesting();
+
+/**
+ * A generic data fetching component that can be used for fetching data for any
+ * component.
+ */
+export default class DataFetcher extends React.Component {
```

Comment:
Also recommending we follow style guide and avoid default export here.

Reply:
Okay! [That part](https://google.github.io/styleguide/jsguide.html#es-module-exports) of the style guide definitely slipped by me, so I'll need to update all of the React components to follow the style guide. I'll do that in another PR, but for now I'll change the DataFetcher from `export default` in this PR.